query-translator/src/translate.rs

```diff
@@ -455,19 +456,43 @@ fn re_project(mut inner: SelectQuery, projection: Projection) -> SelectQuery {
             order: order_by,
             limit: limit,
         };
+    }
 
-    if nullable.is_empty() {
-        outer
-    } else {
-        SelectQuery {
-            distinct: false,
-            projection: Projection::Star,
-            from: FromClause::TableList(TableList(vec![TableOrSubquery::Subquery(Box::new(outer))])),
-            constraints: nullable,
-            group_by: vec![],
-            order: vec![],
-            limit: Limit::None,
-        }
+    // `ORDER BY` is relevant when we have a `LIMIT`.  We therefore lift the `ORDER BY` if there’s
```

Comment:
Worth mentioning that we lift it because SQL doesn't guarantee ordering unless it's at the outermost level.